app/models/concerns/key_pair_compute_resource.rb

```diff
@@ -23,8 +23,12 @@ def recreate
     setup_key_pair
   end
 
-  def delete_key_pair(kay_pair_name)
-    delete_key_from_resource(kay_pair_name)
+  def delete_key_from_resource(remote_key_pair = key_pair)
+    logger.info "removing #{provider_friendly_name}: #{name} key #{remote_key_pair.name}"
+    key = client.key_pairs.get(remote_key_pair.name)
+    key.nil? || key.destroy
+  rescue => e
+    logger.warn "failed to delete key pair from #{provider_friendly_name}: #{name}, you might need to cleanup manually : #{e}"
```

Comment:
I think this should be proper error with stacktrace, do not rethrow tho, but keep it in logs.

        Foreman::Logging.exception("Failed to delete key pair", e)